### DISCUSSION TITLE
Start of the art OCR models?

### DISCUSSION THREAD

**Discussion by innat** [mindee/doctr#948] [General]
By the DocTr, the supported models are mentioned [HERE](https://github.com/mindee/doctr#text-detection). I'm wondering, why there are no state-of-the-art models yet or if they're in the roadmap currently. Some of the models are 

- [ABCNet](https://github.com/aim-uofa/AdelaiDet/blob/master/configs/BAText/README.md)
- [MaskTextSpotter](https://github.com/MhLiao/MaskTextSpotterV3)

**Comment by felixdittrich92** [comment]
Hi @innat :wave:,

excuse the late reply. 
Of course in a later state we will add more models also for text detection but currently we have to focus on some optimizations and to bring the current state into a stable one.
But feel free to open an feature request ticket with your models you think we should add or if you want open a PR we are happy about every addition/help :100: 
About both models one point that would be a problem, is that ABCNet or TextFuseNet and some other are build on top of PyTorch's detectron2 framework i don't see any way to integrate this (I think we will focus on models which can be used/implemented in both frameworks tensorflow and pytorch and don't need some heavy dependencies) :sweat_smile: 

But if you have some models in mind which meet this criteria let us know :smiley:

**Reply by innat** [reply]
> Of course in a later state we will add more models also for text detection but currently we have to focus on some optimizations and to bring the current state into a stable one. But feel free to open a feature request ticket with the models you think we should add or if you want open a PR we are happy about every addition/help

That would be great. 

> About both models one point that would be a problem, is that ABCNet or TextFuseNet and some others are built on top of PyTorch's detectron2 framework I don't see any way to integrate this (I think we will focus on models which can be used/implemented in both frameworks tensorflow and pytorch and don't need some heavy dependencies

The ABCNet has its [own repo](https://github.com/Yuliang-Liu/bezier_curve_text_spotting) earlier but I think it's incomplete and the full code is released in [Adet](https://github.com/aim-uofa/AdelaiDet/blob/master/configs/BAText/README.md). But yes, there is (afaik) no tf implementation currently and needs the effort to translate the code to pytorch to tensorflow/keras.

**Reply by frgfm** [reply]
That is an interesting idea, I agree that for each upcoming release, we should start a discussion with the community to identify valuable new models to integrate :+1:

**Comment by innat** [comment]
@felixdittrich92 @frgfm 
Possibility to adopt [keras-core](https://keras.io/keras_core/)

**Reply by felixT2K** [reply]
Hi @innat 👋🏼 ,

Keras Core is still in beta (first release fall 2023). (Btw. currently on Layers in for pytorch backend are supported not nn.Modules)
At some point i think we need to replace `tf.keras` with `tf.keras_core` for the TF implementations.
If your question is to merge all models PT and TF into one keras_core version we would need to evaluate firstly that it does not break the current implementations and pretrained checkpoints also performance would be a topic to check before.

You could definitely take a look at it, but not before it's signed stable.
